sources/packages/api/src/services/atbc/atbc.service.ts

```diff
@@ -0,0 +1,180 @@
+        {
+          usr: this.config.ATBCUserName,
+          pwd: this.config.ATBCPassword,
+          app: this.config.ATBCApp,
+        },
+        { httpsAgent: agent },
+      );
+      return authRequest?.data as ATBCAuthTokenResponse;
+    } catch (excp) {
+      this.logger.error(`Received exception while ATBC Authentication`);
+      this.logger.error(`${JSON.stringify(excp.response)} `);
+      throw excp;
+    }
+  }
+
+  /**
+   * Create client on ATBC.
+   * @returns the result of a success full authentication or throws an exception
+   * in case the result is anything different from HTTP 200 code.
+   */
+  public async ATBCCreateClient(
+    payload: ATBCCreateClientPayload,
+  ): Promise<ATBCCreateClientResponse> {
+    try {
+      const config = await this.getConfig();
+      const apiEndpoint =
+        this.config.ATBCEndpoint.slice(-1) === "/"
+          ? `${this.config.ATBCEndpoint}pd-clients`
+          : `${this.config.ATBCEndpoint}/pd-clients`;
+      const res = await axios.post(`${apiEndpoint}`, payload, config);
+      return res?.data as ATBCCreateClientResponse;
+    } catch (excp) {
+      this.logger.error(`Received exception while creating client at ATBC`);
+      this.logger.error(excp);
+      throw excp;
+    }
+  }
+  /**
+   * Check PD status of the student with SIN number on ATBC.
+   * @returns the result of a success full authentication or throws an exception
+   * in case the result is anything different from HTTP 200 code.
+   */
+  private async ATBCPDChecker(
+    payload: ATBCPDCheckerPayload,
+  ): Promise<ATBCPDCheckerResponse> {
+    try {
+      const config = await this.getConfig();
+      const apiEndpoint =
+        this.config.ATBCEndpoint.slice(-1) === "/"
+          ? `${this.config.ATBCEndpoint}pd`
+          : `${this.config.ATBCEndpoint}/pd`;
+      const res = await axios.post(`${apiEndpoint}`, payload, config);
+      return res?.data as ATBCPDCheckerResponse;
+    } catch (excp) {
+      this.logger.error(
+        `Received exception while checking the PD status at ATBC`,
+      );
+      this.logger.error(excp);
+      throw excp;
+    }
+  }
+
+  public async PDCheckerProcess(eachStudent: Student) {
+    try {
+      this.logger.log(`Creating Payload for student id ${eachStudent.id}`);
+      // create PD checker payload
+      const payload: ATBCPDCheckerPayload = {
+        id: eachStudent.sin,
+      };
+      // api to check the student PD status in ATBC
+      this.logger.log(`Check for PD status of student ${eachStudent.id}`);
+      // try {
+      const response = await this.ATBCPDChecker(payload);
+      // e9yStatusId === 1 , PD Confirmed
+      // e9yStatusId === 2 , PD Denied
+      // e9yStatusId === 0 , Processing, In Queue
+
+      this.logger.log(
+        `PD Status for student ${eachStudent.id}, status ${response?.e9yStatusId} ${response?.e9yStatus}, `,
+      );
+      if (response?.e9yStatusId === 1 || response?.e9yStatusId === 2) {
```

Comment:
Please avoid doing the comparison with numbers. Considering creating an enum to keep the below codes.
The comments help, but the values should be better expressed in enum/constants.
// e9yStatusId === 1 , PD Confirmed
// e9yStatusId === 2 , PD Denied
// e9yStatusId === 0 , Processing, In Queue